test/end-to-end/core.sh

```diff
@@ -109,6 +109,22 @@ os::cmd::expect_success "curl -f http://${DOCKER_REGISTRY}/healthz"
 os::cmd::expect_success "dig @${API_HOST} docker-registry.default.local. A"
 registry_pod="$(oc get pod -n default -l deploymentconfig=docker-registry --template='{{(index .items 0).metadata.name}}')"
 
+os::log::info "Configure registry to disable mirroring"
+os::cmd::expect_success "oc project '${CLUSTER_ADMIN_CONTEXT}'"
+os::cmd::expect_success 'oc env -n default dc/docker-registry REGISTRY_MIDDLEWARE_REPOSITORY_OPENSHIFT_MIRRORPULLTHROUGH=false'
+os::cmd::expect_success 'oc env -n default dc/docker-registry REGISTRY_MIDDLEWARE_REPOSITORY_OPENSHIFT_PULLTHROUGH=true'
+os::cmd::expect_success 'oc rollout status dc/docker-registry'
+os::log::info "Registry configured to disble mirroring"
+
+# verify local image based on remote image can be pushed to local registry
+os::log::info "Check the image based on a non-local base image."
+os::cmd::expect_success "oc login -u user0 -p pass"
+os::cmd::expect_success "oc new-project project0"
+os::cmd::expect_success "oc new-build -D \$'FROM busybox:glibc\nRUN echo abc >/tmp/foo'"
```

Comment:
@legionus ouch, it works better than I though. I'm sorry for the fuss. I didn't know this.